app/frontend/controllers/reference_verse_controller.js

```diff
@@ -0,0 +1,130 @@
+
+  try {
+    var verses = await ipcNsi.getBookText(currentBibleTranslationId, bookShortTitle, mappedAbsoluteVerseNumber, 1);
+    var verseText = referenceVerseContainer.querySelector('.verse-text');
+    verseText.innerHTML = verses[0].content;
+    app_controller.sword_notes.initForContainer(referenceVerseContainer);
+    verseListController.bindEventsAfterBibleTextLoaded(undefined, false, $(referenceVerseContainer));
+    app_controller.dictionary_controller.bindAfterBibleTextLoaded();
+  } catch (e) {
+    console.warn('Could not update translation for reference verse: ' + e);
+  }
+}
+
+function clearReferenceVerse(tabIndex=undefined) {
+  var currentVerseListFrame = verseListController.getCurrentVerseListFrame(tabIndex);
+  var referenceVerseContainer = currentVerseListFrame[0].querySelector('.reference-verse');
+
+  referenceVerseContainer.innerHTML = '';
+}
+
+async function renderReferenceVerse(verseBox, tabIndex=undefined) {
+  if (verseBox == null || verseBox.length != 1) return;
+
+  var currentVerseListFrame = verseListController.getCurrentVerseListFrame(tabIndex);
+  var currentVerseList = verseListController.getCurrentVerseList(tabIndex);
+  var referenceVerseContainer = currentVerseListFrame[0].querySelector('.reference-verse');
+  var classList = currentVerseList[0].classList;
+
+  for (let i = 0; i < classList.length; i++) {
+    let currentClass = classList[i];
+
+    if (currentClass != "verse-list") {
+      referenceVerseContainer.classList.add(currentClass);
+    }
+  }
+
+  var clonedVerseBox = verseBox[0].cloneNode(true);
+  var header = await verseBoxHelper.getLocalizedVerseReference(verseBox[0]);
+  var referenceVerseHeader = "<div class='reference-header'>" + header + "</div>";
+  referenceVerseContainer.innerHTML = referenceVerseHeader;
+  referenceVerseContainer.appendChild(clonedVerseBox);
+  referenceVerseContainer.innerHTML += "<br/><hr/>";
+
+  var currentTab = app_controller.tab_controller.getTab(tabIndex);
+  var textType = currentTab.getTextType();
+  var textTypeHeader = "";
+
+  if (textType == 'xrefs') {
+    textTypeHeader = `<span i18n="general.module-xrefs">${i18n.t('general.module-xrefs')}</span>`;
+  } else if (textType == 'tagged_verses') {
+    textTypeHeader = `<span i18n="tags.verses-tagged-with">${i18n.t('tags.verses-tagged-with')}</span> <i>${currentTab.getTagTitleList()}</i>`;
+  }
+
+  referenceVerseContainer.innerHTML += "<div class='reference-verse-list-header'><h2>" + textTypeHeader + "</h2></div>";
+  verseListController.bindEventsAfterBibleTextLoaded(undefined, false, $(referenceVerseContainer));
+  app_controller.dictionary_controller.bindAfterBibleTextLoaded(tabIndex);
+}
+
+module.exports = {
+  showReferenceContainer,
+  getCurrentReferenceVerse,
+  getLocalizedReferenceVerse,
+  updateReferenceVerseTranslation,
+  clearReferenceVerse,
+  renderReferenceVerse
+};
```

Comment:
I think, for new modules, there should be consistency with exports. Previously we used 
```javascript
module.exports.renderReferenceVerse = async function (verseBox, tabIndex=undefined) {
```
Instead, we can switch to this rewrite of `module.eports` object (Although some people say that it might be a problem in rear cases, which I don't think we should worry about). It's just should be consistent throughout new/refactored code.

Personally, I like what we used, as it shows which functions/methods are external right away.

Reply:
Thanks! Consistency is important - I agree. I will change to the `module.exports.functionname` ... style.